ddl/ddl_api.go

```diff
@@ -806,6 +806,16 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 			buf := new(bytes.Buffer)
 			s.Partition.Expr.Format(buf)
 			pi.Expr = buf.String()
+			if s.Partition.Tp == model.PartitionTypeRange {
+				if _, ok := s.Partition.Expr.(*ast.ColumnNameExpr); ok {
+					for _, col := range cols {
+						name := strings.Replace(col.Name.String(), ".", "`.`", -1)
+						if fmt.Sprintf("`%s`", name) == pi.Expr && col.Tp != mysql.TypeLong {
+							return errors.Trace(fmt.Errorf("Field '%s' is of a not allowed type for this type of partitioning", pi.Expr))
```

Comment:
There is a example on how to do it:
https://github.com/pingcap/tidb/blob/4b2b4bed2cafc983cee2fd754b607788e15306df/ddl/ddl.go#L574

Error code is important information, that's why we need to define them for MySQL errors.

Reply:
DONE